### ISSUE TITLE
remove vertices in areas that you "cut".

### ISSUE THREAD

**Issue by bowmana** [Zylann/godot_heightmap_plugin#291]
Remove vertices in areas that you "cut", so that there is no collider associated in spots without textures. Also navmesh would bake around said cutouts. I have tried making a separate collider in areas that I have cut out, that when hit it disables the terrain collider. However this seems like a sloppy workaround. Please add this!

**Comment by Zylann** [comment]
Given the technique used by this terrain system, it is not a straightforward thing to do. Physics, rendering and navigation are three different systems that dont know each other.

Rendering does not actually use a mesh for the terrain, it uses a repeated plane displaced in shader. For that there is not even "vertices" to remove, the removal happens per pixel.

For physics to take holes into account, this is already an existing issue #125, which depends on Godot supporting that in the first place in its heightmap collider.

Navigation is a pain and is the only one that needs a giant mesh just to get generated properly. This one could indeed be improved to include holes depending on the shader used (which may or may not support holes, or in different ways). Though it can only be precise per vertices, or has to use marching squares to get closer to the visuals.

So in summary... in terms of use case, you are asking for a feature that's already tracked physics-wise, so the only untracked part is navigation.

**Comment by bowmana** [comment]
I see that makes sense. Thank you!
